### PR TITLE
#2148 added schema for work-package-template and blocked_by_info

### LINKED ISSUE #2148: Work Package Templates - Schema
### Description

We wanna create the schema for the new work package templates feature we're adding to Finishline

### Acceptance Criteria

- The Schema described below has been added to the schema.prisma file.
- A migration has been created from the schema changes (Run yarn prisma:migrate once you're done making changes in schema.prisma) with an appropriate name
- Any associated tests, seed file, etc. are fixed if the changes cause errors in them.

### Proposed Solution

Schema Changes:

#### Blocked_By_Info:
blockedByInfoId                String     uuid()
stage                                 Work_Package_Stage?
name                                 String
workPackageTemplate      Work_Package_Template

#### Work_Package_Template
workPackageTemplateId   String     uuid()
templateName                   String
templateNotes                   String
workPackageName                                 String?
stage                                 Work_Package_Stage?
duration                            Int?
blockedBy                         Blocked_By_Info[]
expectedActivities            String[]
deliverables                      String[]
dateCreated                     Datetime
userCreated                      User
dateDeleted                     Datetime?
userDeleted                      User?

### PR DESCRIPTION
**Pull request by 8q8b**
## Changes

Added schema for work_package_template and blocked_by_info

## Screenshots
image

image

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2148

### CODE REVIEW

**Review by Aaryan1203 — APPROVED** [review]
Great work! LGTM

File: src/backend/src/prisma/schema.prisma [hunk 1]
```diff
@@ -83,49 +83,51 @@ model User {
...
+  requiredDesignReviews        Design_Review[]         @relation(name: "requiredAttendee")
+  optionalDesignReviews        Design_Review[]         @relation(name: "optionalAttendee")
+  userConfirmedDesignReviews   Design_Review[]         @relation(name: "confirmedAttendee")
+  userDeniedDesignReviews      Design_Review[]         @relation(name: "deniedAttendee")
+  attendedDesignReviews        Design_Review[]         @relation(name: "userAttended")
+  createdDesignReviews         Design_Review[]         @relation(name: "designReviewCreator")
+  deletedDesignReviews         Design_Review[]         @relation(name: "designReviewDeleter")
+  drScheduleSettings           Schedule_Settings?
+  createdWork_Package_Template Work_Package_Template[] @relation(name: "workPackageTemplateCreator")
+  deletedWork_Package_Template Work_Package_Template[] @relation(name: "workPackageTemplateDeleter")
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:130]: Shouldn't have the underscores in either of these names

File: src/backend/src/prisma/schema.prisma [hunk 2]
```diff
@@ -636,3 +638,29 @@ model Meeting {
...
+  workPackageTemplate   Work_Package_Template @relation(fields: [workPackageTemplateId], references: [workPackageTemplateId])
+  workPackageTemplateId String
+}
+
+model Work_Package_Template {
+  workPackageTemplateId String             @id @default(uuid())
+  templateName          String
+  templateNotes         String
+  workPackageName       String?
+  stage                 Work_Package_Stage
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:655]: This should also be optional

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/backend/src/prisma/schema.prisma [hunk 3]
```diff
@@ -636,3 +638,29 @@ model Meeting {
...
+  stage                 Work_Package_Stage?
+  name                  String
+  workPackageTemplate   Work_Package_Template @relation(fields: [workPackageTemplateId], references: [workPackageTemplateId])
+  workPackageTemplateId String
+}
+
+model Work_Package_Template {
+  workPackageTemplateId String              @id @default(uuid())
+  templateName          String?
+  templateNotes         String?
```
Comment (Aaryan1203) [inline src/backend/src/prisma/schema.prisma:653]: I don't think these two are supposed to be optional

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Looks good, but can you make it in one migration instead of two, so delete the two migration files and folders and then run yarn:prisma migrate as well

File: src/backend/src/prisma/schema.prisma (hunk 3 above)
Comment (RChandler234) [reply src/backend/src/prisma/schema.prisma:653]: ^^

**Review by Aaryan1203 — APPROVED** [review]
LGTM!

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**